modules/hal_sensors/i2c_sensor_driver/src/I2C_sensor_driver.cpp

```diff
@@ -36,80 +36,84 @@ constexpr uint8_t GetLowerByte(uint16_t number) {
   return (number & 0xFF);
 }
 
-void I2CDriver::Init() {
-  i2c_peripheral_->begin();
+void I2C_sensor_driver::init_i2c_helper(void) {
+  hal::i2c::I2C_Driver::begin();
 }
 
-void I2CDriver::ChangeAddress(uint8_t new_i2c_address) {
-  i2c_addr_ = new_i2c_address;
+void I2C_sensor_driver::ChangeAddress(hal::i2c::I2CAddr new_i2c_address) {
+  this->set_i2c_addr(new_i2c_address);
 }
 
-void I2CDriver::WriteReg(uint16_t reg, uint8_t data) {
+void I2C_sensor_driver::write8_reg16b(uint16_t reg, uint8_t data) {
```

Comment:
I know why you choose the name.. But think twice.. All the registers of the current en future sensors use 16-bit addressing.. So why make the naming worse. You could of  course instead change it to WriteReg8 instead..